### ISSUE TITLE
ESMF error when using LocalDaskExecutor

### ISSUE THREAD

**Issue by norlandrhagen** [carbonplan/cmip6-downscaling#85]
cc @jhamman @orianac 

When running the bcsd flow in the PangeoRuntime, I get the error:

<details>

```python

└── 01:59:10 | ERROR   | Task 'get_spatial_anomalies': Exception encountered during task execution!
Traceback (most recent call last):
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/prefect/engine/task_runner.py", line 876, in get_task_run_state
    value = prefect.utilities.executors.run_task_with_timeout(
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/prefect/utilities/executors.py", line 454, in run_task_with_timeout
    return task.run(*args, **kwargs)  # type: ignore
  File "/home/jovyan/carbonplan/cmip6-downscaling/cmip6_downscaling/methods/bcsd.py", line 261, in get_spatial_anomalies
    obs_interpolated, _ = regrid_dataset(
  File "/home/jovyan/carbonplan/cmip6-downscaling/cmip6_downscaling/workflows/utils.py", line 409, in regrid_dataset
    regridder = xe.Regridder(ds_rechunked, target_grid_ds, "bilinear", extrap_method="nearest_s2d")
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xesmf/frontend.py", line 770, in __init__
    grid_in, shape_in, input_dims = ds_to_ESMFgrid(
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xesmf/frontend.py", line 132, in ds_to_ESMFgrid
    grid = Grid.from_xarray(lon.T, lat.T, periodic=periodic, mask=None)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xesmf/backend.py", line 111, in from_xarray
    grid = cls(
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/ESMF/util/decorators.py", line 81, in new_func
    return func(*args, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/ESMF/api/grid.py", line 410, in __init__
    self._struct = ESMP_GridCreateNoPeriDim(self.max_index,
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/ESMF/interface/cbindings.py", line 581, in ESMP_GridCreateNoPeriDim
    raise ValueError('ESMC_GridCreateNoPeriDim() failed with rc = '+str(rc)+
ValueError: ESMC_GridCreateNoPeriDim() failed with rc = 545.    Please check the log files (named "*ESMF_LogFile").
└── 01:59:10 | INFO    | Task 'get_spatial_anomalies': Finished task run for task with final state: 'Failed'

```
</details>

In L207 of runtimes.py, if `return LocalDaskExecutor(scheduler="threads")` is replaced with `return LocalExecutor()` there is no ESMF error and the flow runs as expected. 

#### Additional Information:
 - The get_spatial_anomalies_task contains the `tags=['dask-resource:TASKSLOTS=1']` tag
 - branch: https://github.com/carbonplan/cmip6-downscaling/tree/feature/add_pyramid_bcsd

**Comment by orianac** [comment]
change `scheduler='processes'`

**Comment by jhamman** [comment]
closed by #86
